### ISSUE TITLE
Work items are not deleted when space is removed

### ISSUE THREAD

**Issue by ljelinkova** [openshiftio/openshift.io#4382]
##### Issue Overview
When I create work items in a space, they are not deleted when I remove the space. 

##### Expected Behaviour
Work items should be deleted when the space is removed. 
##### Current Behaviour
Work items are not deleted when the space is removed.

**Comment by aptmac** [comment]
Duplicate?: https://github.com/openshiftio/openshift.io/issues/4283

**Comment by ppitonak** [comment]
@aptmac yes, it was an intentional duplicate. The original issue was about two bugs so we split them into two issues.

**Comment by muruGanesan** [comment]
@sunilmalagi, @Veethika,
IMO: It looks like an integration issue.
Because, from the UXD perspective, we provide notification when a user is about to delete the space and all the data related to that particular Space is soft-deleted - Is there any deviation in 'Planner' flow?
Could please look at this issue and provide your inputs?

CC: @nimishamukherjee, @michaelkleinhenz

**Comment by Veethika** [comment]
@ljelinkova @ppitonak the mentioned expected behavior in the description is valid. When a space is deleted, all the work items within that space should be removed as well.  Given the very high consequence, a warning should be displayed to the user before they confirm the action for space deletion, stating that all the contained work-items would be removed. 
@michaelkleinhenz @nimishamukherjee ^sounds valid?

**Comment by ppitonak** [comment]
@Veethika you are right that it would be nice to expand the message that user sees when deleting the space. This is how it looks like now:

![osio_delete_space](https://user-images.githubusercontent.com/445100/46601822-ac819200-caee-11e8-9144-b8d9ef98ccfa.png)

**Comment by Veethika** [comment]
@Preeticp Could we come up with a better message for: "Removing this space would remove all the work-items contained within, do you want to proceed?"

**Comment by muruGanesan** [comment]
@Veethika, thanks. 
@Preeticp: We may have to educate the user about:
Removal of "applications, workspace, code base and contributors" - when a user removes the space.

**Comment by Preeticp** [comment]
@muruGanesan @Veethika please see here: https://github.com/openshiftio/openshift.io/issues/2107#issuecomment-362405121
A detailed design and messaging has already been provided by @catrobson.
Also searching for Space deletion showed many other issues with Space deletion which probably needs to be resolved before @catrobson's suggestion is implemented, probably that's why it is on hold.

**Comment by joshuawilson** [comment]
@xcoulon can you explain / share what gets removed when we remove a Space? 

@openshiftio/uxd-team please be specific if you want the UI changed for this. It appears to me as this has nothing to do with the UI but needs to just be fixed in the backend.

**Comment by jarifibrahim** [comment]
>@xcoulon can you explain / share what gets removed when we remove a Space?

We remove the codebase and openshift resources (deployments) associated with space along with the actual space. Currently, we have only soft deletes and due to which all the entities related to space (eg: workitems, area, iterations) are not removed.

**Comment by jarifibrahim** [comment]
Assigning this issue to @kwk because he is working on improving the entity deletion mechanism on the backend.
